### PR TITLE
Action: AWS Get Long Running ElasticCache clusters Without Reserved Nodes

### PR DESCRIPTION
**Pull request by shloka-bhalgat-unskript**
## Description
[Issue-574](https://github.com/unskript/Awesome-CloudOps-Automation/issues/574)

### Testing
Screenshot 2023-05-11 at 9 21 12 PM

### Checklist:
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published. 

### Documentation
Make sure that you have documented corresponding changes in this repository.

### CODE REVIEW

File: AWS/legos/aws_get_long_running_elasticcache_clusters_without_reserved_nodes/aws_get_long_running_elasticcache_clusters_without_reserved_nodes.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    if not region:
+        all_regions = aws_list_all_regions(handle)
+    for reg in all_regions:
+        try:
+            elasticacheClient = handle.client('elasticache', region_name=reg)
+            for cluster in elasticacheClient.describe_cache_clusters()['CacheClusters']:
+                cluster_age = datetime.now(timezone.utc) - cluster['CacheClusterCreateTime']
+                if cluster_age > timedelta(days=threshold):
+                    response = elasticacheClient.describe_reserved_cache_nodes()
+                    if response['ReservedCacheNodes']:
```
Comment (amit-chandak-unskript) [inline AWS/legos/aws_get_long_running_elasticcache_clusters_without_reserved_nodes/aws_get_long_running_elasticcache_clusters_without_reserved_nodes.py:63]: We are getting reserved cache nodes for every cluster?? We should have gotten it once only

**Review by amit-chandak-unskript — CHANGES_REQUESTED** [review]

File: AWS/legos/aws_get_long_running_elasticcache_clusters_without_reserved_nodes/aws_get_long_running_elasticcache_clusters_without_reserved_nodes.py (hunk 1 above)
Comment (shloka-bhalgat-unskript) [reply AWS/legos/aws_get_long_running_elasticcache_clusters_without_reserved_nodes/aws_get_long_running_elasticcache_clusters_without_reserved_nodes.py:63]: We are matching the NodeType of the reserved cluster with the NodeType of the EC cluster.

Comment (amit-chandak-unskript) [reply AWS/legos/aws_get_long_running_elasticcache_clusters_without_reserved_nodes/aws_get_long_running_elasticcache_clusters_without_reserved_nodes.py:63]: @shloka-bhalgat-unskript i got that part, my point is why getting the reserved nodes inside the loop for cluster, that needs to be done only for a region and then use it for all the cluster in that region

File: AWS/legos/aws_get_long_running_elasticcache_clusters_without_reserved_nodes/aws_get_long_running_elasticcache_clusters_without_reserved_nodes.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                if cluster_age > timedelta(days=threshold):
+                    cluster_dict = {}
+                    cluster_dict["region"] = reg
+                    cluster_dict["cluster"] = cluster['CacheClusterId']
+                    cluster_dict["node_type"] = cluster['CacheNodeType']
+                    cluster_data.append(cluster_dict)
+        except Exception:
+            pass
+    if len(node_data) != 0:
+        for node in node_data:
```
Comment (amit-chandak-unskript) [inline AWS/legos/aws_get_long_running_elasticcache_clusters_without_reserved_nodes/aws_get_long_running_elasticcache_clusters_without_reserved_nodes.py:84]: 👎 @shloka-bhalgat-unskript this is wrong logic, imagine 2 regions, one has t1 type reserved node and other has t2 type reserved node. Now, if some 3rd region has t1 type of cluster node, your logic would say it has reserved nodes, though they are in different region. Let me code it up in your branch

**Review by amit-chandak-unskript — APPROVED** [review]

**Pull request merged**